lib/los_santos.rb

```diff
@@ -2,19 +2,18 @@
 require 'yaml'
 
 module LosSantos
-  def self.name_day(date_string = Time.now.strftime('%d-%m'))
-    data = YAML.load_file('lib/los_santos/data/fr.yml')
+  ALLOWED_CLASS = [String, Date, Time, DateTime].freeze
+
+  def self.name_day(date = Time.now)
     begin
-      date_split = date_string.split('-')
-      return data[date_split[1].to_i][date_split[0].to_i]
-    rescue NoMethodError
-      return print_usage
+      date = Date.strptime(date, '%d-%m') if date.class == String
+    rescue ArgumentError
```

Comment:
I think you can remove this begin rescue

Reply:
I put this begin rescue to control the error and display a custom error message

Reply:
I should accept strings like => `2011-05-19` you have quickly conversions like => 

```
date = Date.parse("2011-05-19 10:30:14")
date = Date.parse("2011-05-19")
```

Reply:
I thought to use parse at the beginning, but I don't like her behavior:  
Date.parse('2011-05-19')